test/asm/Makefile.am

```diff
@@ -43,39 +43,40 @@ atomic_barrier_noinline.c:
 	ln -s $(top_srcdir)/test/asm/atomic_barrier.c atomic_barrier_noinline.c
 atomic_barrier_noinline_SOURCES = atomic_barrier_noinline.c
 atomic_barrier_noinline_CFLAGS = $(AM_CFLAGS) -DOMPI_DISABLE_INLINE_ASM
+atomic_barrier_noinline_LDADD = $(libs) -lpthread
```

Comment:
Why are you explicitly `-lpthread`ing everywhere?  I thought the point of this PR was to make threading a framework...?  (i.e., the threading library is therefore variable, right?)

(similar question for including `<pthread.h>` in various tests)

Reply:
This may be a mistake. It should only affect threading libraries that overload the pthreads symbols though. I’ll check when I get to the hotel.

Reply:
I did a check on abstraction leakage wrt to pthreads and there are still a lot of places to get fixed up in both the tests and body of the code, including `ompi/mca/topo/treematch/treematch/` and `ompi/mca/io/romio321/romio/adio/common/`.

I'd argue that we should do one thing at a time. Ompi needs a cleanup in general to get the posixisms out of the non-opal code but that can come after the thread mca is in.

Reply:
Treematch and ROMIO are both third-party / imported codes.

I think it's a fair question to ask how this threads framework is going to address the third part codes in Open MPI (e.g., is libevent also going to be affected / does it call POSIX threads directly?).

Reply:
@jsquyres Libevent will require changes. Right now it relies on preemption. It will need to work with a more tightly integrated yield mechanic that calls to the thread scheduler. That's proriority #1 for after the pull request goes in.

Reply:
we'll fix this.